### ISSUE TITLE
The 5.4.1 connection pool patch created a new connection bug. 

### ISSUE THREAD

**Issue by brady-ss** [logstash-plugins/logstash-integration-jdbc#129]
5.4.1 The plugin does not maintain an accurate connection pool.
The validation option also does not work correctly,
validation query fails to maintain the connection by sending a validation null query with the statement when the scheduler is triggered instead of querying it on its own cycle.

The connection pool was not maintaining the connection when the schedule was run in 10 minute increments, and the second query met below error

> Java::JavaSql:SQLNonTransientConnectionException: unexpected end of stream, read 0 bytes from 4 ( socket was closed by server )

And on the next query, it creates a new connection and the query succeeds.

**Comment by brady-ss** [comment]
this is our input setting

```
    jdbc {
        jdbc_driver_library => "/logstash/template/mariadb-java-client-2.4.4.jar"
        jdbc_driver_class => "org.mariadb.jdbc.Driver"
        jdbc_connection_string => "jdbc:mariadb://x.x.x.x:x/viewname?serverTimezone=Asia/Seoul&sessionVariables=wait_timeout=660"
        jdbc_user => "xxx"
        jdbc_password => "xxx"
        schedule => "1-59/10 * * * *" 
        statement => "SELECT *
                    FROM x
                    WHERE x_dt > :sql_last_value
                    ORDER BY x_dt"
        tracking_column => "x_dt"
        plugin_timezone => "utc"
        jdbc_default_timezone => "Asia/Seoul"
        use_column_value => true
        tracking_column_type => "timestamp"                    
        last_run_metadata_path => "/xxx/xxx.db"
        lowercase_column_names => false
        connection_retry_attempts => 24
        connection_retry_attempts_wait_time => 3600          
        type => "xxx_xxx_xxx"
    }
```

and this is our jdbc_validation test input 

```
    input {
        jdbc {
            jdbc_validate_connection => true
            jdbc_validation_timeout => 30
            sql_log_level => debug
            #jdbc_pool_timeout => 0
            jdbc_driver_library => "/logstash/template/mariadb-java-client-2.4.4.jar"
            jdbc_driver_class => "org.mariadb.jdbc.Driver"
            jdbc_connection_string => "jdbc:mariadb://x.x.x.x:x/viewname?serverTimezone=Asia/Seoul"
            jdbc_user => "xxx"
            jdbc_password => "xxx"
            schedule => "1-59/10 * * * *" 
            statement => "SELECT *
                        FROM x
                        WHERE x_dt > :sql_last_value
                        ORDER BY x_dt"
            tracking_column => "x_dt"
            plugin_timezone => "utc"
            jdbc_default_timezone => "Asia/Seoul"
            use_column_value => true
            tracking_column_type => "timestamp"                    
            last_run_metadata_path => "/xxx/xxx.db"
            lowercase_column_names => false
            connection_retry_attempts => 24
            connection_retry_attempts_wait_time => 3600          
            type => "xxx_xxx_xxx"
        }
    }
```

The validation query does not work independently of the schedule and runs at the exact same time as the schedule, so it cannot act as a validation.

and this is debug level log 


```
2023-05-10 18:01:00.330	logstash	DEBUG	(0.007014s) SELECT xxx
    FROM xxx
    WHERE xxx > '2023-05-10 17:49:05.000000'
    ORDER BY xxx	/logstash/logs_test/pipeline_TEST_XXX.log
2023-05-10 18:01:00.321	logstash	DEBUG	(0.002465s) SELECT NULL	/logstash/logs_test/pipeline_TEST_XXX.log
2023-05-10 18:11:01.092	logstash	DEBUG	(0.002897s) SELECT xxx
    FROM xxx
    WHERE xxx > '2023-05-10 18:00:07.000000'
    ORDER BY xxx	/logstash/logs_test/pipeline_TEST_XXX.log
2023-05-10 18:11:01.027	logstash	ERROR	Java::JavaSql::SQLNonTransientConnectionException: (conn=337689) unexpected end of stream, read 0 bytes from 4 (socket was closed by server): SELECT NULL	/sw_log/sca/logstash/logs_test/pipeline_SCA_TEST_xxx.log

```


validation query "select null" 2023-05-10 18:01:00.321
our main query at 2023-05-10 18:01:00.321

You can even see that the validation query encounters a socket close error in the following query at 2023-05-10 18:11:01.027

we solved this problem by connection string option "wait_timeout=660"
But even in this case, I'm still getting random disconnections.

----

To summarize, there are two issues

1. the connection pool is not working properly and I am losing connections every time, this was not happening before the connection pool patch.
2. the validation query option is not working properly because it is synchronized with schedule.

**Comment by brady-ss** [comment]
https://github.com/logstash-plugins/logstash-integration-jdbc/issues/143
